server/src/test/java/org/elasticsearch/cluster/metadata/MetadataIndexTemplateServiceTests.java

```diff
@@ -1062,6 +1064,83 @@ public void testPutExistingComposableTemplateIsNoop() throws Exception {
         assertThat(metadataIndexTemplateService.addIndexTemplateV2(state, false, "foo", template), equalTo(state));
     }
 
+    public void testUnreferencedDataStreamsWhenAddingTemplate() throws Exception {
```

Comment:
maybe add a unit test that tests with a cluster state with no templates, but a few data streams and then
adding a template shoudl still work. This to simulate the restore scenario?

Reply:
:+1: I re-organized this test so that it has that behavior